starlette/middleware/cors.py

```diff
@@ -66,12 +73,22 @@ def __call__(self, scope: Scope):
                 if method == "OPTIONS" and "access-control-request-method" in headers:
                     return self.preflight_response(request_headers=headers)
                 else:
-                    return functools.partial(
-                        self.simple_response, scope=scope, origin=origin
-                    )
+                    if self.is_allowed_origin(origin=origin):
+                        return functools.partial(
+                            self.simple_response, scope=scope, origin=origin
+                        )
+                    return PlainTextResponse("Disallowed CORS origin", status_code=400)
```

Comment:
An interesting point to make here: We *don't* currently disallow the requests on the server side, as CORS is enforced by the browser rather than by the server.

We *could* choose to implement server-side enforcement *too*, but it'd involve a little more work than *just* checking the origin, since we'd also want to eg. check if allowed headers have been sent, check if allow_credentials is being respected etc.etc.

I'm not against us doing that, or doing that as an optional point of configuration, but let's treat it as a separate issue, since there's more to consider than just origin checks.